app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -225,18 +153,38 @@ export default class AMRHeatmapVis extends React.Component {
     }
   }
 
+  findAMRCountForName(rowName, sample) {
+    const { selectedOptions } = this.props;
+    switch (selectedOptions.viewLevel) {
+      case VIEW_LEVEL_ALLELES: {
+        return sample.amrCounts.find(
+          amrCount => amrCount["allele"] === rowName
+        );
+      }
+      case VIEW_LEVEL_GENES: {
+        let amrCountForRow = sample.amrCounts.find(
+          amrCount => amrCount["annotation_gene"] === rowName
```

Comment:
Can you add a comment somewhere explaining the difference between an `annotation_gene` and a `gene`?

Reply:
So it's kind of technical, but what gets passed as the gene name by the pipeline is a locus. A locus is a site in a DNA or RNA sequence where a cluster of different alleles may be found, but perform essentially the same function. When srst2 identifies an AMR gene, it compares a sequence to all the reference sequences that match to the same locus and gives only 1 best match between all the reference sequences. This is so that srst2 does not report closely related sequences as different genes. The locus name is what is used by the pipeline for "gene", and that's what gets stored under the `gene` key for an `AmrCount` in a sample.

However, the ARG ANNOT database also defines a gene name in the annotation for each reference sequence. This gene name is *much* more closely aligned with how genes are listed in CARD. I believe the issue of naming AMR genes is something that's still in the process of being solved by the research community. I could report the locus name as well (and in fact I think I'll update this to add that).

Reply:
Added the above as a comment

Reply:
This is super helpful! Now that I realize how deep the explanation is, I think this paragraph would be better to put in your AMR documentation file, so that we don't have large technical explanations in the middle of the code. Thanks for writing this up.

I do have a question (which we can discuss offline). So the gene is the "locus", and the annotation_gene is the reference sequence which had the best match. I would have expected the different reference sequences to be "alleles". So what's the difference between an allele and an annotation_gene?